src/service/cid-api.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getCookie, setCookie} from '../cookies';
+import {Services} from '../services';
+import {dev} from '../log';
+import {dict} from '../utils/object';
+
+const GOOGLE_API_URL = 'https://ampcid.google.com/v1/publisher:getClientId?key=';
+const API_KEYS = {
+  'googleanalytics': 'AIzaSyA65lEHUEizIsNtlbNo-l2K18dT680nsaM',
```

Comment:
Where is this API key used? on viewer or on origin? on origin, would it help to have a different API key to distinguish traffic between analytics.js and amp-analytics?